File: 백준/Silver/11047. 동전 0/동전 0.py
def solve(N, K):
    rst = 0
    for i in range(N-1, -1, -1):
        while K >= 0:
            if data[i] <= K:
                K -= data[i]
                rst += 1
            else:
                break
    return rst


N, K = map(int, input().split())
data = [int(input()) for _ in range(N)]

# print(N)
# print(K)
# print(data)

print(solve(N, K))

File: 백준/Silver/11047. 동전 0/동전 0.js
let fs = require('fs');
let input = fs.readFileSync('/dev/stdin').toString().split('\n');

let n = Number(input[0].split(' ')[0])
let price = Number(input[0].split(' ')[1])

let coin = input.slice(1).map(Number)

let count = 0
for (i=n-1; i>=0; i--) {
  if (coin[i] <= price) {
    while (coin[i] <= price){
      price -= coin[i]
      count += 1
    }
  }
}

console.log(count)